addons/project/models/project_project_stage.py

```diff
@@ -41,6 +42,20 @@ def unlink_wizard(self, stage_view=False):
         }
 
     def write(self, vals):
+        if vals.get('company_id'):
+            # Checking if there is a project with a different company_id than the target one. If so raise an error since this is not allowed
+            project = self.env['project.project'].search(['&', ('stage_id', 'in', self.ids), ('company_id', '!=', vals['company_id'])], limit=1)
+            if project:
+                company = self.env['res.company'].browse(vals['company_id'])
```

Comment:
Can't `vals['company_id']` be False? In that case, the error message will be a little :hankey:

Reply:
It can be, but then this portion of code won't be executed thanks to `if vals.get('company_id')`